src/scoping-shim.js

```diff
@@ -259,17 +267,25 @@ export default class ScopingShim {
     if (!this._isRootOwner(host)) {
       this._elementsHaveApplied = true;
     }
-    if (overrideProps) {
-      styleInfo.overrideStyleProperties =
-        styleInfo.overrideStyleProperties || {};
-      Object.assign(styleInfo.overrideStyleProperties, overrideProps);
-    }
+    this._mixOverrideStyleProps(styleInfo, overrideProps);
     if (!nativeCssVariables) {
       this.styleElementShimVariables(host, styleInfo);
     } else {
       this.styleElementNativeVariables(host, styleInfo);
     }
   }
+  /**
+   * @param {StyleInfo} styleInfo
+   * @param {Object=} overrideProps
+   */
+  _mixOverrideStyleProps(styleInfo, overrideProps) {
+    if (!styleInfo || !overrideProps) {
```

Comment:
Let's move these conditionals out of here. You should only need the `overrideProps` check at https://github.com/webcomponents/shadycss/pull/237/files#diff-a15f14a51481bba582abcc3f00fad9d0R270

Reply:
Done.